interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodecompilation/NodeCompiler.scala

```diff
@@ -149,35 +150,68 @@ class NodeCompiler(
+      def withFragmentParameterErrors[T](
+          compiledObject: ValidatedNel[ProcessCompilationError, T],
+          validationContext: ValidationContext
+      ) = {
+        val paramValidationErrors = params.flatMap { param =>
+          FragmentParameterValidator.validate(param, id, this, validationContext)
+        }
+
+        compiledObject.andThen(v =>
+          NonEmptyList.fromList(paramValidationErrors) match {
+            case Some(errors) => Invalid(errors)
+            case None         => Valid(v)
+          }
+        )
+      }
+
       definitions.sourceFactories.get(id) match {
         case Some(definition) =>
           val parameters                           = fragmentDefinitionExtractor.extractParametersDefinition(frag).value
           val variables: Map[String, TypingResult] = parameters.map(a => a.name -> a.typ).toMap
-          val validationContext = Valid(
-            contextWithOnlyGlobalVariables.copy(localVariables =
-              contextWithOnlyGlobalVariables.globalVariables ++ variables
-            )
+          val validationContext = contextWithOnlyGlobalVariables.copy(localVariables =
+            contextWithOnlyGlobalVariables.globalVariables ++ variables
           )
 
-          compileObjectWithTransformation[Source](
+          val compilationResult = compileObjectWithTransformation[Source](
             Nil,
             Nil,
             Left(contextWithOnlyGlobalVariables),
             None,
             definition,
-            _ => validationContext
+            _ => Valid(validationContext)
           ).map(_._1)
+
+          compilationResult.copy(compiledObject =
+            withFragmentParameterErrors(compilationResult.compiledObject, validationContext)
+          )
+
         case None =>
-          NodeCompilationResult(
-            Map.empty,
-            None,
-            Valid(
-              contextWithOnlyGlobalVariables.copy(localVariables =
-                params.map(p => p.name -> loadFromParameter(p)).toMap
+          loadParametersTypeMap(params) match {
+            case Valid(paramTypeMap) =>
+              val validationContext =
+                contextWithOnlyGlobalVariables.copy(localVariables = paramTypeMap)
+
+              val compilationResult = NodeCompilationResult(
+                Map.empty,
+                None,
+                Valid(validationContext),
+                Valid(new StubbedFragmentInputTestSource(frag, fragmentDefinitionExtractor).createSource())
               )
-            ),
-            Valid(new StubbedFragmentInputTestSource(frag, fragmentDefinitionExtractor).createSource())
-          )
+
+              compilationResult.copy(compiledObject =
+                withFragmentParameterErrors(compilationResult.compiledObject, validationContext)
+              )
+
+            case Invalid(errors) =>
+              NodeCompilationResult(
+                Map.empty,
+                None,
+                Invalid(errors),
```

Comment:
I'm not sure if errors should be duplicated? `NodeCompilationResult` has `def errors()` method that concatenates these fields, so each error would be twice. Is this desired?

Reply:
ye i've encountered this issue, but it's either this or
1. return valid validationContext or valid compiledObject - not really possible if the FragmentInputDefinition is completely bad
2. deduplicate errors in `def errors()`, doesn't feel right either